File: frontend/src/types/report.ts
export type InsightTone = "positive" | "neutral" | "attention";

export interface Insight {
  title: string;
  detail: string;
  tone: InsightTone;
}

export interface KpiItem {
  label: string;
  value: string;
}

export interface ReportSummary {
  file_id: string;
  file_name: string;
  row_count: number;
  column_count: number;
  kpis: KpiItem[];
  insights: Insight[];
}

export type ReportFormat = "pdf" | "excel" | "csv";
export type ReportType = "PDF" | "EXCEL" | "CSV";

export interface ReportRecord {
  id: string;
  title: string;
  type: ReportType;
  createdAt: string;
  uploadedFile: { originalName: string } | null;
}

File: backend/app/schemas/report.py
from typing import Literal

from pydantic import BaseModel

InsightTone = Literal["positive", "neutral", "attention"]


class Insight(BaseModel):
    title: str
    detail: str
    tone: InsightTone


class KpiItem(BaseModel):
    label: str
    value: str


class ReportSummary(BaseModel):
    file_id: str
    file_name: str
    row_count: int
    column_count: int
    kpis: list[KpiItem]
    insights: list[Insight]


ReportFormat = Literal["pdf", "excel", "csv"]
